<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>MapDisto Genetics Software</title>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.css">
  <link rel="stylesheet" href="../style.css">
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css" rel="stylesheet"
    integrity="sha384-+0n0xVW2eSR5OomGNYDnhzAbDsOXxcvSN1TPprVMTNDbiYZCxYbOOl7+AMvyTG2x" crossorigin="anonymous">
  <link rel="icon" href="../img/fi.png">
</head>

<body class="fw-light">

  <!-- ######################################### navbar ######################################## -->

  <nav class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
    <div class="container">
      <a class="navbar-brand fw-light fs-4" href="../index.html">
        MapDisto Genetics Software
      </a>

      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
        aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>

      <div class="collapse navbar-collapse fw-light fs-4" id="navbarNav">
        <ul class="navbar-nav ms-auto">

          <li class="nav-item">
            <a class="nav-link" href="downloads.html">Downloads</a>
          </li>

          <li class="nav-item">
            <a class="nav-link" href="blog.html">Blog</a>
          </li>

          <li class="nav-item">
            <a class="nav-link" href="resources.html">Resources</a>
          </li>

          <li class="nav-item">
            <a class="nav-link" href="../index.html#contact">Contact</a>
          </li>

          <li class="nav-item">
            <a class="nav-link active" aria-current="page" href="../index.html">
              <svg xmlns="http://www.w3.org/2000/svg" width="26" height="26" fill="currentColor"
                class="bi bi-house pb-1" viewBox="0 0 16 16">
                <path fill-rule="evenodd"
                  d="M2 13.5V7h1v6.5a.5.5 0 0 0 .5.5h9a.5.5 0 0 0 .5-.5V7h1v6.5a1.5 1.5 0 0 1-1.5 1.5h-9A1.5 1.5 0 0 1 2 13.5zm11-11V6l-2-2V2.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5z" />
                <path fill-rule="evenodd"
                  d="M7.293 1.5a1 1 0 0 1 1.414 0l6.647 6.646a.5.5 0 0 1-.708.708L8 2.207 1.354 8.854a.5.5 0 1 1-.708-.708L7.293 1.5z" />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>

  <div class="container bg-secondary my-4 p-4 rounded text-center fs-5">
    <h2 class="display-6 text-center my-4 py-4">Blog</h2>
  </div>

  <div class="container my-4 py-4 text-center bg-secondary rounded">
    <div class="row row-cols-1 row-cols-md-2 g-4 ">
      <div class="col">
        <div class="card">
          <img src="../img/profil.jpg" class="card-img-top" alt="...">
          <div class="card-body">
            <h4 class="card-title">Mapdisto</h4>
            <p class="card-text">MapDisto is a program for mapping genetic markers and quantitative trait loci (QTL) in
              experimental segregating populations. MapDisto is very easy to use. It has many features such as finding
              linkage groups, drawing genetic maps, computing allelic desequilibrium (segregation distortion), automatic
              ordering of loci, QTL analysis, exporting data and maps for other mapping software, etc... One of its
              specificities is to propose recombination fraction estimates in case of segregation distortion.
              Supported population types are backcross, F2, doubled haploids, RIL-SSD (recombinant lines by single-seed
              descent), and HRIL (highly recombinant lines). MapDisto has been developed in Microsoft Visual Basic and
              needs any Microsoft Excel to run. It works well on Windows and Mac OS X platforms.</p>
          </div>

          <div class="accordion" id="accordionPanelsStayOpenExample">
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingOne">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapseOne" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapseOne">
                  New Features
                </button>
              </h2>
              <div id="panelsStayOpen-collapseOne" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingOne">
                <div class="accordion-body">
                  <p>New features in MapDisto 2.x include:</p>
                  <ul class="text-start mt-2">
                    <li>handling of very big genotyping datasets like the ones generated by genotyping-by-sequencing
                      (GBS), through a Java plugin by Christopher Heffelfinger</li>
                    <li>direct importation of Variant Call Format (VCF) files</li>
                    <li>construction of linkage groups in case of segregation distortion</li>
                    <li>data imputation on VCF files using a combination of LB-Impute (Fragoso et al, Genetics 2015) and
                      BP-Impute (Fragoso et al 2017, in prep)</li>
                    <li>QTL search via new R/qtl interface</li>
                    <li>meta-QTL analysis in multiple populations sharing a common parent</li>
                    <li>local recombination analysis</li>
                  </ul>
                </div>
              </div>
            </div>
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingFive">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapseFive" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapseFive">
                  Features
                </button>
              </h2>
              <div id="panelsStayOpen-collapseFive" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingFive">
                <div class="accordion-body">
                  <p>Efficient software such as Mapmaker or Joinmap have been developed to construct genetic maps from
                    linkage data of experimental segregating populations. However, no option is proposed in these
                    programs to deal with genetic markers which show deviations from expected Mendelian frequencies in
                    their segregation ratios.
                    Moreover, the use of these programs is not always intuitive and to learn how to deal with their
                    commands can be time-consuming. MapDisto has tools to:</p>
                  <ul class="text-start mt-2">
                    <li>Compute and draw genetic maps easily and quickly through a graphical interface</li>
                    <li>Facilitate the analysis of marker data showing segregation distortion due to differential
                      viability of gametes or zygotes</li>
                    <li>Find linkage groups, with specified minimum LOD score and maximum recombination frequency</li>
                    <li>Automatically order loci on linkage groups using one of the three available algorithms</li>
                    <li>Compute tables of recombination fractions (various estimates), map distances in centimorgans,
                      linkage and independence khi2s, two-point LOD scores, correlations, for all couples of loci in a
                      sequence, [G-tests for F2 - LODs with SD]</li>
                    <li>Compute segregation khi2s that measure the deviation from expected Mendelian segregation and
                      their associated probabilities for all loci of a particular sequence</li>
                    <li>Compute several map parameters for a specified sequence</li>
                    <li>Look for genotyping errors and correct them</li>
                    <li>Draw genetic maps computed either by MapDisto or imported from another program</li>
                    <li>Find QTLs using one-way ANOVA and display a graphical output</li>
                    <li>Export data and run (.in) files for MapMaker/EXP</li>
                    <li>Export computed maps and data for advanced QTL analysis programs such as QTL Cartographer,
                      WinQTLCart, QGene, or MapQTL</li>
                    <li>Run automated commands that make the computation of an entire genetic map very fast and easy
                    </li>
                    <li>Compare maps</li>
                    <li>And many other things</li>
                  </ul>
                  <p>MapDisto has automated commands that make the computation of an entire genetic map very fast and
                    easy.
                    The graphical interface permits us to manage data and to navigate into the different output
                    datasheets.
                    A help that describes MapDisto features and the main steps to follow for data analysis is also
                    available.
                    The present version of the program handles for several types of segregating populations: backcross
                    (BC1), doubled haploid (DH), single-seed descent (SSD), F2, highly-recombinant inbred lines (HRIL).
                    A build-in help is included. However, it is very basic. For more help, please read the tutorial.
                    MapDisto has been developed in Microsoft Visual Basic and needs any Microsoft Excel to run. It works
                    well on Windows and Mac OS X platforms.</p>
                </div>
              </div>
            </div>
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingTwo">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapseTwo" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapseTwo">
                  Screenshots
                </button>
              </h2>
              <div id="panelsStayOpen-collapseTwo" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingTwo">
                <div class="accordion-body">
                </div>
              </div>
            </div>
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingThree">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapsgThree" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapsgThree">
                  Citations
                </button>
              </h2>
              <div id="panelsStayOpen-collapsgThree" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingThree">
                <div class="accordion-body">
                  You can perform a <a class="link-dark"
                    href="http://scholar.google.com/scholar?hl=en&q=mapdisto&btnG=&as_sdt=1%2C5&as_sdtp=">citation
                    search</a> in Google Scholar.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>

      <div class="col">
        <div class="card">
          <img src="../img/profil.jpg" class="card-img-top" alt="...">
          <div class="card-body">
            <h4 class="card-title">CSSL Finder</h4>
            <p class="card-text">CSSL Finder is a program for managing introgression lines, and particularly useful to
              develop Chromosome Segment Substitution Lines (CSSLs). Thanks to its graphical interface, CSSL Finder is
              very easy to use. It can:
            <ul class="text-start">
              <li>Search a population of advanced backcrossed lines for a set of lines that optimizes the representation
                of the donor genome
              </li>
              <li>Display graphical genotypes of the lines
              </li>
              <li>Display graphical phenotypic values of the lines</li>
            </ul>
            </p>
          </div>

          <div class="accordion" id="accordionPanelsStayOpenExample2">
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingOne2">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapseOne2" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapseOne2">
                  Features
                </button>
              </h2>
              <div id="panelsStayOpen-collapseOne2" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingOne">
                <div class="accordion-body">
                  CSSL Finder aims to help at developing, managing and analyzing advanced introgression lines, like
                  Chromosome Segment Substitution Lines, or CSSLs. The main features of this program are:
                  <ul class="text-start mt-2">
                    <li>find the best set of lines that maximizes the donor genome coverage, with the smallest
                      undesirable genetic background</li>
                    <li>display graphical genotypes of this set</li>
                    <li>display graphical phenotypic values of these lines</li>
                    <li>look for QTLs through ANOVA 1 analysis (including permutation tests for empirical threshold
                      determination)</li>
                    <li>obtain statistics on lines and markers</li>
                  </ul>
                  Several other features are available, that you will discover while playing with the different options.
                </div>
              </div>
            </div>
            <div class="accordion-item">
              <h2 class="accordion-header" id="panelsStayOpen-headingTwo2">
                <button class="accordion-button collapsed text-dark fs-5" type="button" data-bs-toggle="collapse"
                  data-bs-target="#panelsStayOpen-collapseTwo2" aria-expanded="false"
                  aria-controls="panelsStayOpen-collapseTwo2">
                  Screenshots
                </button>
              </h2>
              <div id="panelsStayOpen-collapseTwo2" class="accordion-collapse collapse"
                aria-labelledby="panelsStayOpen-headingTwo2">
                <div class="accordion-body">
                </div>
              </div>
            </div>
          </div>
          
        </div>
      </div>

      <div class="col" id="noisy">
        <div class="card">
          <img src="../img/profil.jpg" class="card-img-top" alt="...">
          <div class="card-body">
            <h4 class="card-title">NOISYmputer</h4>
            <p class="card-text">Low-coverage next-generation sequencing (LC-NGS) methods can be used to genotype
              bi-parental populations. This approach allows creating highly saturated genetic maps at reasonable cost,
              precisely localized recombination breakpoints, and minimize mapping intervals for quantitative-trait locus
              analysis.
              The main issues with these genotyping methods are (1) poor performance at heterozygous loci, (2) a high
              percentage of missing data, (3) local errors due to erroneous mapping of sequencing reads and reference
              genome mistakes, and (4) global, technical errors inherent to NGS itself.
              Recent methods like Tassel-FSFHap or LB-Impute are excellent at addressing issues 1 and 2, but nonetheless
              perform poorly when issues 3 and 4 are persistent in a dataset (i.e. “noisy” data).
              We present NOISYmputer, program for imputation of LC-NGS data that eliminates the need of complex
              pre-filtering of noisy data, accurately types heterozygous chromosomic regions, corrects erroneous data,
              and imputes missing data.
              NOISYmputer performs very well for noisy data, compared with Tassel-FSFHap, LB-Impute, or
              Genotype-Corrector.</p>
          </div>
        </div>

      </div>
    </div>
  </div>

  <!-- ######################################### footer-social-section ######################################## -->

  <div class="container bg-secondary rounded">
    <div class="row py-4 justify-content-start">
      <div class="col-xl-2 col-lg-3 col-md-4 ms-4">
        <a class="link-dark fs-5" href="https://www.facebook.com/MapDisto" target="_blank">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-facebook me-2"
            viewBox="0 0 16 16">
            <path
              d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z" />
          </svg>Facebook page</a>
      </div>

      <div class="col-xl-2 col-lg-2 ms-4">
        <a class="link-dark fs-5" href="blog.html">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-book me-2"
            viewBox="0 0 16 16">
            <path
              d="M1 2.828c.885-.37 2.154-.769 3.388-.893 1.33-.134 2.458.063 3.112.752v9.746c-.935-.53-2.12-.603-3.213-.493-1.18.12-2.37.461-3.287.811V2.828zm7.5-.141c.654-.689 1.782-.886 3.112-.752 1.234.124 2.503.523 3.388.893v9.923c-.918-.35-2.107-.692-3.287-.81-1.094-.111-2.278-.039-3.213.492V2.687zM8 1.783C7.015.936 5.587.81 4.287.94c-1.514.153-3.042.672-3.994 1.105A.5.5 0 0 0 0 2.5v11a.5.5 0 0 0 .707.455c.882-.4 2.303-.881 3.68-1.02 1.409-.142 2.59.087 3.223.877a.5.5 0 0 0 .78 0c.633-.79 1.814-1.019 3.222-.877 1.378.139 2.8.62 3.681 1.02A.5.5 0 0 0 16 13.5v-11a.5.5 0 0 0-.293-.455c-.952-.433-2.48-.952-3.994-1.105C10.413.809 8.985.936 8 1.783z" />
          </svg>Check our blog</a>
      </div>

      <div class="col-xl-2 col-lg-3 ms-4">
        <a class="link-dark fs-5" href="resources.html#author">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-code me-2"
            viewBox="0 0 16 16">
            <path
              d="M5.854 4.854a.5.5 0 1 0-.708-.708l-3.5 3.5a.5.5 0 0 0 0 .708l3.5 3.5a.5.5 0 0 0 .708-.708L2.707 8l3.147-3.146zm4.292 0a.5.5 0 0 1 .708-.708l3.5 3.5a.5.5 0 0 1 0 .708l-3.5 3.5a.5.5 0 0 1-.708-.708L13.293 8l-3.147-3.146z" />
          </svg>About Mathias</a>
      </div>

      <div class="col-xl-3 col-lg-3 ms-4">
        <a class="link-dark fs-5" href="https://groups.google.com/u/1/g/mapdisto" target="_blank">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor" class="bi bi-google me-2"
            viewBox="0 0 16 16">
            <path
              d="M15.545 6.558a9.42 9.42 0 0 1 .139 1.626c0 2.434-.87 4.492-2.384 5.885h.002C11.978 15.292 10.158 16 8 16A8 8 0 1 1 8 0a7.689 7.689 0 0 1 5.352 2.082l-2.284 2.284A4.347 4.347 0 0 0 8 3.166c-2.087 0-3.86 1.408-4.492 3.304a4.792 4.792 0 0 0 0 3.063h.003c.635 1.893 2.405 3.301 4.492 3.301 1.078 0 2.004-.276 2.722-.764h-.003a3.702 3.702 0 0 0 1.599-2.431H8v-3.08h7.545z" />
          </svg>Discussion group</a>
      </div>
    </div>

    <div class="row py-4 justify-content-end">
      <div class="col-lg-3 col-md-4 col-sm-5 text-end">
        &copy; 2021 sahvanart
        <a class="link-dark" href="mailto:sahvanart@gmail.com?subject=Contact%20from%20Mapdisto%20Software%20website">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor"
            class="bi bi-envelope-fill mb-1 ms-1" viewBox="0 0 20 20">
            <path
              d="M.05 3.555A2 2 0 0 1 2 2h12a2 2 0 0 1 1.95 1.555L8 8.414.05 3.555zM0 4.697v7.104l5.803-3.558L0 4.697zM6.761 8.83l-6.57 4.027A2 2 0 0 0 2 14h12a2 2 0 0 0 1.808-1.144l-6.57-4.027L8 9.586l-1.239-.757zm3.436-.586L16 11.801V4.697l-5.803 3.546z" />
          </svg>
        </a>
        <a class="link-dark" href="https://github.com/sahvanart/mapdisto">
          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="currentColor"
            class="bi bi-github mb-1 ms-1" viewBox="0 0 20 20">
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z" />
          </svg>
        </a>
      </div>
    </div>
  </div>


  <!-- JavaScript and dependencies -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js"
    integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous">
  </script>

</body>

</html>